### PR TITLE
TabContainerModel can now persist the active tab.

### PR DESCRIPTION
**Pull request by cnrudd**
This PR implements this issue: https://github.com/xh/hoist-react/issues/1980

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes: no breaking changes.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile: works on mobile!
- [x] [Created Toolbox branch](https://github.com/xh/toolbox/tree/persistWithOnTabModel)

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: cmp/tab/TabContainerModel.js [hunk 1]
```diff
@@ -86,27 +88,18 @@ export class TabContainerModel {
                 return;
             }
 
+            if (persistWith) {
+                console.warn('Active tab persistance is ignored when a tab container uses routing.');
```
Comment (lbwexler) [inline cmp/tab/TabContainerModel.js:92]: makes sense -- perhaps mention in the docstring, for persistWith.  Also spelling of `persistence`

File: cmp/tab/TabContainerModel.js [hunk 2]
```diff
@@ -86,27 +88,18 @@ export class TabContainerModel {
...
+
             this.addReaction({
                 track: () => XH.routerState,
                 run: this.syncWithRouter
             });
 
             this.forwardRouterToTab(this.activeTabId);
-        }
-
-        if (track) {
```
Comment (lbwexler) [inline cmp/tab/TabContainerModel.js:97]: Was this removed by accident?

File: cmp/tab/TabContainerModel.js [hunk 3]
```diff
@@ -215,4 +208,41 @@ export class TabContainerModel {
 
         return null;
     }
+
+    setupStateProvider(persistWith) {
+        // Read state from provider -- fail gently
+        let state = null;
+        if (persistWith) {
```
Comment (lbwexler) [inline cmp/tab/TabContainerModel.js:215]: we know this is truthy -- only calling it when truthy in constructor

File: cmp/tab/TabContainerModel.js (hunk 2 above)
Comment (cnrudd) [reply cmp/tab/TabContainerModel.js:97]: Yes!  Sorry!   Fixed.

File: cmp/tab/TabContainerModel.js (hunk 1 above)
Comment (cnrudd) [reply cmp/tab/TabContainerModel.js]: Changed to throwIf up top as suggested.  Spelling no longer an issue.

File: cmp/tab/TabContainerModel.js (hunk 3 above)
Comment (cnrudd) [reply cmp/tab/TabContainerModel.js]: Correct.   I have removed the if block.

**Pull request merged**